packages/jitm/src/JITM.php

```diff
@@ -149,6 +153,152 @@ static function jitm_jetpack_woo_services_activate( $CTA ) {
 		);
 	}
 
+	/**
+	 * This is an entire admin notice dedicated to messaging and handling of the case where a user is trying to delete
+	 * the connection owner.
+	 */
+	function delete_user_update_connection_owner_notice() {
+		global $current_screen;
+
+		if ( ! isset( $current_screen->base ) || 'users' !== $current_screen->base ) {
+			return;
+		}
+
+		if ( ! isset( $_REQUEST['action'] ) || 'delete' !== $_REQUEST['action'] ) {
+			return;
+		}
+
+		// Get connection owner or bail
+		$connection_manager = new Manager();
+		$connection_owner   = $connection_manager->get_connection_owner();
+		if ( ! $connection_owner ) {
+			return;
+		}
+
+		// Bail if we're not trying to delete connection owner.
+		$user_ids_to_delete = array();
+		if ( isset( $_REQUEST['users'] ) ) {
+			$user_ids_to_delete = $_REQUEST['users'];
+		} elseif ( isset( $_REQUEST['user'] ) ) {
+			$user_ids_to_delete[] = $_REQUEST['user'];
+		}
+		$deleting_connection_owner = in_array( $connection_owner->ID, (array) $user_ids_to_delete );
+		if ( ! $deleting_connection_owner ) {
+			return;
+		}
+
+		// Bail if they're trying to delete themselves to avoid confusion.
+		if ( $connection_owner->ID == get_current_user_id() ) {
+			return;
+		}
+
+		$connection_manager = new Manager();
+		$connected_users    = $connection_manager->get_connected_users( 'jetpack_connect' );
+
+		echo "<div class='notice notice-warning' id='jetpack-notice-switch-connection-owner'>";
+		echo '<h2>' . esc_html__( 'Important notice about your Jetpack connection:', 'jetpack' ) . '</h2>';
+		echo '<p>' . sprintf( esc_html__( "Warning! You are about to delete the Jetpack connection owner (%s) for this site, which may cause some of your Jetpack features to stop working.", 'jetpack' ),
+			esc_html( $connection_owner->data->user_login )
+		) . '</p>';
+
+		if ( ! empty( $connected_users ) && count( $connected_users ) > 1 ) {
+			echo '<form id="jp-switch-connection-owner" action="" method="post">';
+			echo "<label for='owner'>" . esc_html__( 'You can choose to transfer connection ownership to one of these already-connected admins:', 'jetpack' ) . ' </label>';
```

Comment:
Not a blocker, but I think in a future PR we should also mention something about the plan, when the site has one. We want to let folks know that the plan will remain attached the the previous owner.

Reply:
Only checking for `isset` may not be enough here since `get_userdata( $connection_owner_id )` will always return at least `false`. How about this instead:

```suggestion
			is_a( $connection_owner_userdata, 'WP_User' ) ? esc_html( $connection_owner_userdata->data->user_login ) : ''
```